app/models/mentorship.rb

```diff
@@ -44,16 +44,31 @@ def send_mail
 
 private
 
+  # def get_possible_mentors
+  #   if mentee.stage_of_career == 5
+  #     User.where(mentor: true, waitlist: false, stage_of_career: 5)
+  #         .where( "mentor_times > ?", 0)
+  #         .where(mentor_industry: mentee.primary_industry)
+  #         .where("id != ?", mentee.id)
+  #   else
+  #     User.where(mentor: true, waitlist: false)
+  #         .where("stage_of_career > ? AND mentor_times > ?", mentee.stage_of_career, 0)
+  #         .where(mentor_industry: mentee.primary_industry)
+  #   end
+  # end
+
   def get_possible_mentors
+    industry_key = MentorIndustry.find_by_name(mentee.primary_industry).id
+
     if mentee.stage_of_career == 5
       User.where(mentor: true, waitlist: false, stage_of_career: 5)
           .where( "mentor_times > ?", 0)
-          .where(mentor_industry: mentee.primary_industry)
-          .where("id != ?", mentee.id)
+          .where.not(id: mentee.id)
+          .joins(:mentor_industry_users).where("mentor_industry_users.mentor_industry_id" => industry_key)
     else
       User.where(mentor: true, waitlist: false)
           .where("stage_of_career > ? AND mentor_times > ?", mentee.stage_of_career, 0)
```

Comment:
I found a bug while manually testing. We should not be checking the mentor's stage_of_career, but instead the career_stage on mentor_industry_users. Please add a test that covers this scenario, something like a user has their stage of career as 1, but is mentoring for another industry with a stage of 5.